docs/en/toolkit/env_config_ref.md

```diff
@@ -0,0 +1,342 @@
+
+## File locations 
+
+Within your Pipeline Configuration, the `config/` directory contains all the files and folders that are meant to be customized. In `config/`, you’ll find three sub-directories: `cache`, `core`, and `env`. The `env` directory holds the environment configuration files, so this document will refer to the files in `config/env`.
+
+![env Folder Contents](./images/env_config_ref/1.png)
+
+In the Default Configuration, you’ll see the following files in `config/env/`:
+
+```
+asset.yml
+asset_step.yml
+project.yml
+sequence.yml
+shot.yml
+shot_step.yml
+```
+
+Each of these files corresponds to an environment; having separate files allows each environment to be configured separately.
+
+## How Toolkit determines the current environment
+
+Toolkit uses a core hook called [pick_environment](https://github.com/shotgunsoftware/tk-core/blob/master/hooks/pick_environment.py) to determine which environment file to use at a given time. The `pick_environment` hook’s return values correspond to environment configuration files. For example, if `pick_environment` returns `shot_step`, Toolkit will use `config/env/shot_step.yml` to configure the Toolkit environment.
+
+## Custom environments
+
+The environment configuration files listed above are the ones provided with the Default Configuration. However, some studios may want to employ different or additional environments. For example, a studio might want different configuration settings for every stage in the pipeline – `asset_step_rig`, `asset_step_model`, `shot_step_anim`, `shot_step_light`, and so on. Fortunately, you can fully customize the available environments. 
+
+To do so, you would add the desired environment config files to the `config/env` directory, and override the `pick_environment` core hook, adding to it the logic that defines when to use your new environments. 
+
+## Basic structure
+
+Toolkit’s configuration files are written in [YAML](https://yaml.org/). The general configuration structure for any bundle (a *bundle* is an app, engine, or framework) is as follows:
+
+```
+bundle_name:
+  setting1: value
+  setting2: value
+  complex_setting:
+    sub_setting1: value  
+    Sub_setting2: value
+  location: 
+    type: descriptor_type
+    descriptor_setting1: value
+    descriptor_setting2: value
+```
+
+Bundles can be nested; the value of a setting for one bundle may itself be another bundle. Specifically, as you’ll see in the example below, every engine has an `apps` setting, and its value is the list of apps that are defined for that engine, along with each of the app’s own settings.
+
+Let’s start with a very simple example: an environment in which you have a single engine, with a single app defined within it. The following might be the contents of `project.yml` in this config:
```

Comment:
Probably being very picky here, but the "Let's start" bit feels more like guide language rather than reference. There isn't a continuation of steps after this example, as "The engines block" is not a sub heading of "Basic structure"

Reply:
I changed the wording a bit so it doesn't sound so guide-ish